### DISCUSSION TITLE
How to migrate an application from one applicationSet to an other without recreating resources

### DISCUSSION THREAD

**Discussion by benjamin-bergia** [argoproj/argo-cd#11413] [Q&A]
Following some changes to the structure of our CI, I would like to move applications to new applicationSets. I am currently running ArgoCD v2.5.0 and whatever I try, I always endup losing all my running resources.

My current process is:
- Create a new AS that generates the exact same App than my current AS.
- Wait for ArgoCD to create the new AS and complain about the resources being managed by an other AS.
- Manually delete the Apps on the ArgoCD UI using the "non-cascading" option.
- The Apps get deleted and recreated under the new AS.
- None of my Apps resources were touched.

(Until this point everything is fine.)

- Remove the App from the old AS.

And then ArgoCD deletes all the Apps that I migrated to the new AS with cascading on. (That's usually the point where I reflect on my life and career choices... :sweat_smile: )

I forgot to mention but my applicationSets are managed by ArgoCD itself. So is there any way to migrate my Apps from one AS to the other? And bonus question is it possible to remove cascading deletion by default on all applications an applicationSets?

**Comment by kostis-codefresh** [comment]
How about the following?



1. Check out the Git repo that contains the old AS
2. Change the AS file to have the new structure
3. Commit/push back
4. Wait for Argo CD to sync and in theory no applications will be redeployed if the end result is exactly the same between the two AS

Have you tried this and it didn't work?

**Reply by benjamin-bergia** [reply]
Unfortunately that doesn't work either since I use a different config for the Apps in both AS.

I ended up doing the following:
- update the original AS to add `/spec/syncPolicy/preserveResourcesOnDeletion: true`
- delete the original AS
- create the new AS

**Reply by algo7** [reply]
Does the new AS attempt to re-create those resources or that it detected that that the resources exists already and just showed they everything is in sync?

**Reply by benjamin-bergia** [reply]
Unfortunately, my setup changed quite a bit since then so I can't really reproduce this.

**Reply by keliansb** [reply]
> Does the new AS attempt to re-create those resources or that it detected that that the resources exists already and just showed they everything is in sync?

I can confirm that the new ApplicationSet detects that the resources already exist and displays the Application as **Synced**

**Comment by Jerrimikkihvatai** [comment]
I confirm that setting `/spec/syncPolicy/preserveResourcesOnDeletion: true` prevents app resources from being deleted (ArgoCD v2.14.15).
Mind these things:
1) Note that `/spec/syncPolicy/preserveResourcesOnDeletion: true` goes for ApplicationSet syncPolicy, not Application template syncPolicy
2) You have to delete finalizer in your applicationset as it mentioned [here](https://github.com/argoproj/argo-cd/issues/18201#issuecomment-2525465213) (it was exactly my case why both apps and resources were deleted after appset deletion; If both finalizer and preserveResourcesOnDeletion are defined, your resources will be deleted regardless to preserveResourcesOnDeletion: true)
3) Apps that managed by appset will be deleted, but theirs k8s resources will preserve. Newly created apps will catch orphaned resources
4) Make some tests with dummy apps before moving critical apps from one appst to another
